### ISSUE TITLE
[charts][docs] LineChart stackblitz link creates a broken application

### ISSUE THREAD

**Issue by Janpot** [mui/mui-x#13612]
### Steps to reproduce

Link to live example: https://stackblitz.com/run?file=Demo.tsx

Steps:
1. Open https://mui.com/x/react-charts/line-demo/#simplelinechart
2. Click the stackblitz link for the first demo

### Current behavior

![Screenshot 2024-06-24 at 13 45 48](https://github.com/mui/mui-x/assets/2109932/a62db6d3-524a-42aa-98cc-c65fe8ebb0bd)

### Expected behavior

A working stackblitz application

### Context

Discovered while reporting https://github.com/mui/mui-x/issues/13611

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: stackblitz

**Comment by michelengelen** [comment]
Thanks @Janpot 👍🏼

**Comment by alexfauquette** [comment]
Not sure it's something we can manage on our side. I got a quick look but did not found how to even access to the package effectively installed on the stackblitz demo
https://github.com/stackblitz/core/issues/2530

**Comment by Janpot** [comment]
I downloaded it locally and tried with https://github.com/mui/mui-x/pull/16603 (`https://pkg.csb.dev/mui/mui-x/commit/2216437f/@mui/x-charts`) and it seems to work with that version. (had to also move `tsconfig` from `src` to top-level, but that seems to be unrelated)

**Comment by Janpot** [comment]
Checking https://next.mui.com/x/react-charts/line-demo/#simplelinechart, it looks like there is a new error:

![Image](https://github.com/user-attachments/assets/4bc786e0-bd60-4c03-817a-e7e88d1a0c0d)

This is an error I've seen many times before when multiple React versions are on the page. In fact, it's visible in that error location `/turbo_modules/react@18.3.1/cjs/react.development.js` that it's in react 18, and the devtools confirm that both 18 and 19 are loaded. 

![Image](https://github.com/user-attachments/assets/1150269e-1716-4d11-9f2c-dd50ec5818a5)

I suspect this is caused by `react-spring` peer dependencies.

**Comment by alexfauquette** [comment]
The new error should be fixed when react-spring got removed ;)

**Comment by bernardobelchior** [comment]
Yeah, we're in the process of removing react-spring ([PR](https://github.com/mui/mui-x/pull/17123)). Unless there's a major issue, we should be able to test if this is fixed soon.

**Comment by bernardobelchior** [comment]
With the release of 8.0.0-beta.3, I just tested this and it seems to be working. @Janpot can you confirm?

**Comment by Janpot** [comment]
Can confirm it works for https://next.mui.com/x/react-charts/#overview
